lib/all_your_base_cases.rb

```diff
@@ -37,8 +41,51 @@ def error_message
   end
 end
 
+class AllYourBaseCase::PreProcessor
+  def initialize(row:, index:)
+    @row = row
+    @index = index
+  end
+
+  def run
+    row.merge('expected' => expected_value, 'index' => index)
+  end
+
+  private
+
+  attr_reader :row, :index
+
+  def expected_value
+    return row['expected'] unless row['expected'].nil?
```

Comment:
Which do you prefer?

Personally, I like the fact that it's resilient to the test data changing how it is, but I can go both ways. It does feel a bit strange that some of the exercise logic is reproduced in the generator, though.